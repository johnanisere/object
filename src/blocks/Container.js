import styled from "styled-components";

const Container = styled.section`
  display: flex;
  padding: 20px 10px;
  @media (max-width: 768px) {
    flex-direction: column;
  }
`;
export default Container;
