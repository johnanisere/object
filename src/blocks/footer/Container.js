import styled from "styled-components";

const Container = styled.div`
  -webkit-tap-highlight-color: transparent;
  box-sizing: border-box;
  text-align: ${({ align }) => align || "left"};
  flex: ${({ flex }) => (flex ? flex : 1)};
`;

export default Container;
