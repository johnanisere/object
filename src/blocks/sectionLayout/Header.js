import styled from "styled-components";

const Header = styled.header`
  font-size: 0.8em;
  height:50px;
  box-sizing: border-box;
  font-weight: 600;
  line-height: 1.5;
  color:white;
  padding: 10px 25px;
  display:flex;
  align-items:center;
  justify-content:flex-start;
  border-bottom:2px solid #1e1f2f;
  span{
    display: flex;
    justify-content: center;
    align-items: center;
    svg{
      margin-right:5px;
    }
  }
`;

export default Header;
