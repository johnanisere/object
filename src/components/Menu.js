import React from "react"
import Menu from "../blocks/menu"
import { FaExpand, FaEllipsisH, FaRegHandPaper, FaSistrix } from "react-icons/fa";

const App = () => <Menu>
    <span className="active">
        <FaExpand />
    </span>
    <span className="more">
        <FaEllipsisH />
    </span>
    <span>
        <FaSistrix />
    </span>
    <span>
        <FaRegHandPaper />
    </span>

</Menu>

export default App