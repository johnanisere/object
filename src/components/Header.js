import React from "react";
import PropTypes from "prop-types";
import Header from "../blocks/header";
import { MdKeyboardArrowLeft, MdKeyboardArrowRight } from "react-icons/md"

const App = ({ content }) => {

  return (
    <Header>
      <Header.Content flex={3}>
        <span>
          Project Object Detection | Pascal test 2
      </span>
        <span className="navigation">
          <button><MdKeyboardArrowLeft size={15} /> {` previous`}</button>
          <span className="name">{content}</span>
          <button>next <MdKeyboardArrowRight size={15} /></button>
        </span>
      </Header.Content>
      <Header.Content flex={1}>
        <button className="navigation space">
          skip image <MdKeyboardArrowRight size={15} /><MdKeyboardArrowRight size={15} />
        </button>
        <button className="navigation">
          export
        </button>
      </Header.Content>
    </Header>
  );
};

App.propTypes = {
  content: PropTypes.string.isRequired
};

App.defaultProps = {
  content: "2007_00065.jpg"
};


export default App