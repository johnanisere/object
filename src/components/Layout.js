import React from "react";
import PropTypes from "prop-types";


import Menu from "./Menu"
import Header from "./Header";
import Layout from "../blocks/layout";
import Container from "../blocks/RouteContainer";

const App = ({ children, }) => {

  return (
    <Layout>
      <Header />
      <Layout.Main>
        <Menu />
        <Container>
          {children}
        </Container>
      </Layout.Main>
    </Layout>
  );
};
Layout.propTypes = {
  children: PropTypes.node.isRequired
};

export default App;
