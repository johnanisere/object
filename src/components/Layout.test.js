
import React from "react";
import { shallow } from "enzyme";

import Menu from "./Menu"
import Header from "./Header";
import Layout from "./Layout";
import Container from "../blocks/RouteContainer";
import LayoutContainer from "../blocks/layout";

it("Layout component exists", () => {
    const wrapper = shallow(<Layout />);
    expect(wrapper.exists()).toBe(true);
});

it("Layout component renders", () => {
    const wrapper = shallow(<Layout />);
    expect(wrapper).toMatchSnapshot();
});

it(`Layout should contain "LayoutContainer",  "LayoutContainer.Main", `, () => {
    const wrapper = shallow(<Layout />);

    expect(wrapper.find(Menu)).toHaveLength(1);
    expect(wrapper.find(Header)).toHaveLength(1);
    expect(wrapper.find(Container)).toHaveLength(1);
    expect(wrapper.find(LayoutContainer)).toHaveLength(1);
    expect(wrapper.find(LayoutContainer.Main)).toHaveLength(1);
});