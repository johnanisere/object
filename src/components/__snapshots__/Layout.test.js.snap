// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout component renders 1`] = `ShallowWrapper {}`;
